python/python/bystro/proteomics/annotation_interface.py

```diff
@@ -21,15 +22,563 @@
+
+    Args:
+        document (dict[str, Any]): The document to process.
+
+    Returns:
+        dict[str, Any]: A dictionary with the desired structures of arrays.
+    """
+    result = {}
+
+    def transform_object(obj):
+        return obj
+
+    def traverse_and_transform(obj: dict[str, Any], current_path=""):
+        has_nested_objects = False
+
+        for key, value in obj.items():
+            new_path = f"{current_path}.{key}" if current_path else key
+
+            if isinstance(value, dict):
+                has_nested_objects = True
+                traverse_and_transform(value, new_path)
+
+        if not has_nested_objects or current_path:
+            result[current_path] = transform_object(obj)
+
+    traverse_and_transform(document)
+
+    all_keys = document.keys()
+    for key in all_keys:
+        if key not in result:
+            result[key] = document[key]
+
+    for key in list(result.keys()):
+        nested_keys = [k for k in result if k.startswith(f"{key}.")]
+        for nested_key in nested_keys:
+            sub_key = nested_key[len(key) + 1 :]
+            if sub_key in result[key]:
+                del result[key][sub_key]
+
+    return result
+
+
+def sort_keys(result: dict[str, Any], drop_alt: bool = False) -> list[str]:
+    """
+    Sort keys in a dictionary.
+
+    Args:
+        result (dict[str, Any]): The dictionary to sort keys for.
+        drop_alt (bool): Whether to drop the "alt" key from the sorted keys.
+
+    Returns:
+        list[str]: A sorted list of keys.
+    """
+    keys = list(result.keys())
+
+    if drop_alt and "alt" in keys:
+        keys.remove("alt")
+
+    keys.sort(key=str.lower)
+
+    if "id" in keys:
+        keys.remove("id")
+        keys.insert(0, "id")
+
+    return keys
+
+
+def track_of_objects_to_track_of_arrays(
+    data: Any, track_name: str = "", not_numeric_fields: list[str] | None = None
+) -> Any:
+    """
+    Convert a track of objects to a track of arrays.
+
+    Args:
+        data (Any): The track of objects to convert.
+        track_name (str): The name of the track.
+        not_numeric_fields (list[str] | None):
+            A list of fields that are not numeric, defaults to DEFAULT_NOT_NUMERIC_FIELDS.
+
+    Returns:
+        Any: The converted track of arrays.
+    """
+    if not_numeric_fields is None:
+        not_numeric_fields = DEFAULT_NOT_NUMERIC_FIELDS
+
+    def convert_and_sort(obj, convert_key=""):
+        if obj is None:
+            return None
+
+        if not isinstance(obj, (dict, list)):
+            if convert_key not in not_numeric_fields:
+                num = obj
+
+                if convert_key not in not_numeric_fields and _looks_like_float(obj):
+                    num = float(obj)
+                if not isinstance(num, (int, float)):
+                    return obj
+                if num == 0:
+                    return 0
+                abs_num = abs(num)
+                if abs_num < 0.0001 or abs_num > 1000000:
```

Comment:
What is this range of numbers meant to represent?

Reply:
Good catch; this was a carryover from the javascript code that formatted large numbers in scientific notation, to make displaying those values in cards easier/more readable